numcodecs/tests/test_zarr3.py

```diff
@@ -8,7 +8,7 @@
 if not TYPE_CHECKING:
     zarr = pytest.importorskip("zarr")
 else:
```

Comment:
Would it make sense to remove the `not` here to make this a bit clearer?

Also same question about moving the coverage pragma to the branch

```suggestion
if TYPE_CHECKING:  # pragma: no cover
    import zarr
else:
    zarr = pytest.importorskip("zarr")
```